model/fn-execution/src/main/proto/beam_fn_api.proto

```diff
@@ -257,6 +262,122 @@ message ProcessBundleProgressRequest {
   string instruction_reference = 1;
 }
 
+message MonitoringInfo {
+  // The name defining the metric or monitored state.
+  string urn = 1;
+
+  // This is specified as a URN that implies:
+  // A message class: (Distribution, Counter, Extrema, MonitoringDataTable).
+  // Sub types like field formats - int64, double, string.
+  // Aggregation methods - SUM, LATEST, TOP-N, BOTTOM-N, DISTRIBUTION
+  // valid values are:
+  // beam:metrics:[SumInt64|LatestInt64|Top-NInt64|Bottom-NInt64|
+  //     SumDouble|LatestDouble|Top-NDouble|Bottom-NDouble|DistributionInt64|
+  //     DistributionDouble|MonitoringDataTable]
+  string type = 2;
+
+  // The Metric or monitored state.
+  oneof monitoring_status {
+    MonitoringTableData monitored_table_data = 3;
```

Comment:
Weren't we going to merge these two, branching on type?

Reply:
No, we agreed to distinguish Metrics (The type of monitoring information compatible with metrics collection systems such as Drop Wizard and Stackdriver) from MonitoredState (Other relevant information to collect for debugging a pipeline, such as a Table of File states which indicate why a pipeline is stuck).

So theMmetrics are all in the same group, so its clear that these can be used with systems such as Stackdriver and DropWizard.